oci-objectstorage/src/main/scala/tamer/oci/objectstorage/OciObjectStorageConfiguration.scala

```diff
@@ -0,0 +1,49 @@
+package tamer.oci.objectstorage
+
+import com.sksamuel.avro4s.Codec
+import tamer.{SourceConfiguration, TamerError}
+import zio.URIO
+import zio.stream.ZTransducer
+
+import scala.util.hashing.MurmurHash3.stringHash
+
+trait ObjectNameBuilder[-S] {
+  def startAfter(state: S): Option[String]
+  def objectName(state: S): Option[String]
+}
+
+object OciObjectStorageConfiguration {
+  case class State[R, K, V, S](
+      initialState: S,
+      getNextState: (S, Option[String]) => URIO[R, S],
+      deriveKafkaRecordKey: (S, V) => K
+  )
+}
```

Comment:
this should probably move back in common as the "core" configuration all frontends need, kind of merged back with `SourceConfiguration`. Not for now, but would love to see s3 follow and then some cleanup